core/model/modx/modelement.class.php

```diff
@@ -224,17 +224,18 @@ protected function isStaticFilesAutomated()
      */
     public function getTag() {
         if (empty($this->_tag)) {
-            $propTemp = array();
             if (empty($this->_propertyString) && !empty($this->_properties)) {
+                $propTemp = array();
                 foreach ($this->_properties as $key => $value) {
+                    $key = trim($key);
                     if (is_scalar($value)) {
-                        $propTemp[] = trim($key) . '=`' . $value . '`';
-                    }
-                    else {
-                        $propTemp[] = trim($key) . '=`' . md5(uniqid(rand(), true)) . '`';
+                        $propTemp[$key] = $key . '=`' . $value . '`';
+                    } else {
+                        $propTemp[$key] = $key . '=`' . (is_array($value) ? json_encode($value) : md5(uniqid(rand(), true))) . '`';
```

Comment:
I think it would be better to use `md5(serialize($value))` here as there is no reason to risk json_encode failures (for instance, if encountering a non-UTF8 character). It would also be much more compact in the cache as a hashed string.

Reply:
Yes, you are right, serialize is better. I wanted to switch to it, but was busy a little. I’ll make the changes to my PR.